crates/optimism/rpc/src/sequencer.rs

```diff
@@ -46,54 +47,32 @@ impl SequencerClient {
         &self.inner.http_client
     }
 
-    /// Returns the next id for the request
-    fn next_request_id(&self) -> usize {
-        self.inner.id.fetch_add(1, atomic::Ordering::SeqCst)
-    }
-
-    /// Helper function to get body of the request with the given params array.
-    fn request_body(&self, method: &str, params: Value) -> serde_json::Result<String> {
-        let request = json!({
-            "jsonrpc": "2.0",
-            "method": method,
-            "params": params,
-            "id": self.next_request_id()
-        });
-
-        serde_json::to_string(&request)
-    }
-
-    /// Sends a POST request to the sequencer endpoint.
-    async fn post_request(&self, body: String) -> Result<(), reqwest::Error> {
-        self.http_client()
-            .post(self.endpoint())
-            .header(reqwest::header::CONTENT_TYPE, "application/json")
-            .body(body)
-            .send()
-            .await?;
+    /// Sends a [`alloy_rpc_client::RpcCall`] request to the sequencer endpoint.
+    async fn send_rpc_call(&self, method: &str, params: Value) -> Result<(), SequencerClientError> {
+        self.http_client().request::<Value, ()>(method.to_string(), params).await.inspect_err(
```

Comment:
I'd like to change the params argument here to accept RpcSend, like fn request does, then we don't need to do any `serde_json::Value` prep

Reply:
I agree with that.
Right now it takes Value, but since it is wrapping Alloy’s RpcClient, could type it as impl RpcSend (or even make it generic over T: RpcSend). This would remove the need for json!() everywhere and make things a bit safer and more ergonomic for future method calls.
Could help simplify spots like json!([format!("0x{}", hex::encode(tx))]) into just (encoded_tx,), and aligns with Alloy's usage patterns.

Reply:
done, removed Value and changed  it to accept RpcSend , It's better aligned to `request` now :)